src/main/java/picard/analysis/CollectWgsMetrics.java

```diff
@@ -136,68 +129,241 @@
+         * @param basesExcludedByPairing the number of bases that were filtered out because they were in reads without a mapped mate pair.
+         * @param basesExcludedByBaseq the number of aligned bases that were filtered out because they were of low base quality.
+         * @param basesExcludedByOverlap the number of aligned bases that were filtered out because they were the second observation from an insert with overlapping reads.
+         * @param basesExcludedByCapping the number of aligned bases that were filtered out because they would have raised coverage above the capped value.
+         * @param coverageCap Treat positions with coverage exceeding this value as if they had coverage at this value.
+         * @param baseQHistogram the count of bases observed with a given quality.
+         * @param sampleSize the sample size used for theoretical het sensitivity sampling.
+         */
+        public WgsMetrics(final Histogram<Integer> depthHistogram,
+                          final long basesExcludedByMapq,
+                          final long basesExcludedByDupes,
+                          final long basesExcludedByPairing,
+                          final long basesExcludedByBaseq,
+                          final long basesExcludedByOverlap,
+                          final long basesExcludedByCapping,
+                          final int coverageCap,
+                          final Histogram<Integer> baseQHistogram,
+                          final int sampleSize) {
+            this.depthHistogram = depthHistogram;
+            this.baseQHistogram = baseQHistogram;
+            this.coverageCap    = coverageCap;
+            this.sampleSize     = sampleSize;
+
+            final double total              = depthHistogram.getSum();
+            final double totalWithExcludes  = total + basesExcludedByDupes + basesExcludedByMapq + basesExcludedByPairing + basesExcludedByBaseq + basesExcludedByOverlap + basesExcludedByCapping;
+
+            if (0 < totalWithExcludes) PCT_EXC_MAPQ     = basesExcludedByMapq         / totalWithExcludes;
+            if (0 < totalWithExcludes) PCT_EXC_DUPE     = basesExcludedByDupes        / totalWithExcludes;
+            if (0 < totalWithExcludes) PCT_EXC_UNPAIRED = basesExcludedByPairing      / totalWithExcludes;
+            if (0 < totalWithExcludes) PCT_EXC_BASEQ    = basesExcludedByBaseq        / totalWithExcludes;
+            if (0 < totalWithExcludes) PCT_EXC_OVERLAP  = basesExcludedByOverlap      / totalWithExcludes;
+            if (0 < totalWithExcludes) PCT_EXC_CAPPED   = basesExcludedByCapping      / totalWithExcludes;
+            if (0 < totalWithExcludes) PCT_EXC_TOTAL    = (totalWithExcludes - total) / totalWithExcludes;
+
+            calculateDerivedFields();
+        }
 
         /** The number of non-N bases in the genome reference over which coverage will be evaluated. */
+        @NoMergingIsDerived
```

Comment:
it doesn't have to be, but it clarifies (perhaps) that the different teritorries have to be non-overlapping